src/vellum/workflows/nodes/utils.py

```diff
@@ -124,9 +128,16 @@ def parse_type_from_str(result_as_str: str, output_type: Any) -> Any:
     if issubclass(output_type, BaseModel):
         try:
             data = json.loads(result_as_str)
+            # If we got a FunctionCallVellumValue extract FunctionCall,
+            if (
+                hasattr(output_type, "__name__")
+                and output_type.__name__ == "FunctionCall"
+                and isinstance(data, dict)
+                and "value" in data
+            ):
```

Comment:
[oos] we're going to want to generalize these two checks to the full schema. I was about to write a comment on how to do so, but it got gnarlier than expected, so happy to defer while we get this change in first